.github/workflows/build_and_test.yaml

```diff
@@ -0,0 +1,41 @@
+name: Build and Test
+on:
+  push:
+  pull_request:
+env:
+  GO_VERSION: 1.18.2
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
```

Comment:
Use `cache` from `setup-go`?
https://github.com/actions/setup-go#caching-dependency-files-and-build-outputs

Reply:
Well, it hadn't been released as of yesterday when I did this, but now it has: https://github.com/actions/setup-go/releases/tag/v3.2.0

I'll try it out.

Reply:
Looked good in testing but I'm going to hold off on updating until the `v3` tag is updated to point to this latest setup-go release, so we're not pinned to a specific version of the action and automatically get any patch releases.

Reply:
Alright, they've updated the v3 tag so made this change.